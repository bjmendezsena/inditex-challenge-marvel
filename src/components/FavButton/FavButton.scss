@import "@/styles/variables.scss";

.FavButton {
  appearance: none;
  border: none;
  background-color: transparent;
  cursor: pointer;
  padding: 0;
  margin: 0;
  background-color: transparent;
  color: transparent;
  transition: all 0.3s ease;
  stroke: white;
  stroke-width: 2px;
  width: min-content;
  height: min-content;
  &--active {
    color: $primary-color;
    stroke: transparent;
  }
  &:disabled {
    cursor: not-allowed;
  }
}
