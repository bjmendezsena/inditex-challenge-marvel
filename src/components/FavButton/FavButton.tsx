import clsx from "clsx";
import { ComponentProps } from "react";
import { FavIcon, IconSize } from "@/components";
import "./FavButton.scss";

export type FavButtonProps = {
  isActive?: boolean;
  size?: IconSize;
} & Omit<ComponentProps<"button">, "children">;

export const FavButton = ({
  isActive,
  size,
  className,
  ...props
}: FavButtonProps) => {
  return (
    <button
      {...props}
      className={clsx(
        "FavButton",
        { "FavButton--active": isActive },
        className
      )}
    >
      <FavIcon size={size} />
    </button>
  );
};
