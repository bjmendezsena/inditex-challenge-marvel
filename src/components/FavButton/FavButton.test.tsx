
import { render } from "@testing-library/react";
import { FavButton, FavButtonProps } from "./FavButton";

const defaultProps: FavButtonProps = {
  isActive: true,
};

describe(`<${FavButton.name} />`, () => {
  const getComponent = (props: Partial<FavButtonProps> = {}) => {
    const allProps = { ...defaultProps, ...props };
    return render(<FavButton {...allProps} />);
  };
  it("should render without crashing", () => {
    const { asFragment } = getComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with isActive", () => {
    const { asFragment } = getComponent({ isActive: true });

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render without isActive", () => {
    const { asFragment } = getComponent({ isActive: false });

    expect(asFragment()).toMatchSnapshot();
  });
});
