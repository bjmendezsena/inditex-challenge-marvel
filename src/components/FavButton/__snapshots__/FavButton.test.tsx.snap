// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<FavButton /> > should render with isActive 1`] = `
<DocumentFragment>
  <button
    class="FavButton FavButton--active"
  >
    <svg
      class="icon icon__size--md"
      fill="none"
      viewBox="0 0 56 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`<FavButton /> > should render without crashing 1`] = `
<DocumentFragment>
  <button
    class="FavButton FavButton--active"
  >
    <svg
      class="icon icon__size--md"
      fill="none"
      viewBox="0 0 56 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`<FavButton /> > should render without isActive 1`] = `
<DocumentFragment>
  <button
    class="FavButton"
  >
    <svg
      class="icon icon__size--md"
      fill="none"
      viewBox="0 0 56 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`<FavButton /> should render with isActive 1`] = `
<DocumentFragment>
  <button
    class="FavButton FavButton--active"
  >
    <svg
      class="icon icon__size--md"
      fill="none"
      viewBox="0 0 56 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`<FavButton /> should render without crashing 1`] = `
<DocumentFragment>
  <button
    class="FavButton FavButton--active"
  >
    <svg
      class="icon icon__size--md"
      fill="none"
      viewBox="0 0 56 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`<FavButton /> should render without isActive 1`] = `
<DocumentFragment>
  <button
    class="FavButton"
  >
    <svg
      class="icon icon__size--md"
      fill="none"
      viewBox="0 0 56 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;
